### PR TITLE
minor updates to read boxlib data

### PR DESCRIPTION
**Pull request by luminosa42**
The update makes yt compatible to read in the newly-formatted output from boxlib/amrex plot files, which uses mainly by Orion, Nyx, Maestro and Castro.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
Ping @atmyers

File: yt/frontends/boxlib/1 [hunk 1]
```diff
@@ -0,0 +1,86 @@
+class AMReXParticleHeader(object):
```
Comment (atmyers) [inline yt/frontends/boxlib/1]: Could you remove this file from the PR? I think all of this is already in `data_structures.py`

File: yt/frontends/boxlib/data_structures.py [hunk 2]
```diff
@@ -1225,7 +1225,7 @@ class NyxDataset(BoxlibDataset):
 
     def __init__(self, output_dir,
                  cparam_filename='inputs',
-                 fparam_filename='probin',
+                 fparam_filename='None',
```
Comment (atmyers) [inline yt/frontends/boxlib/data_structures.py]: Could you make this be `None` (without quotes) instead of the string `'None'`?

**Comment by MatthewTurk** [comment]
Thanks for your pull request!

**Comment by zingale** [comment]
I agree, this should be good for Maestro and Castro as well -- we should never need an inputs or probin

**Comment by ngoldbaum** [comment]
LGTM, @atmyers can you hit the merge button?

**Comment by atmyers** [comment]
luminosa42 If you want to include the changes to Castro and Maestro in this PR, I'll hold off on merging until you're ready. If not, I can include those changes in a separate PR.

**Comment by luminosa42** [comment]
@atmyers I've no experience with Castro and Maestro, so I guess I'll leave that to you?

**Comment by ngoldbaum** [comment]
Thank you for your first contribution to yt! 🎆 💯 🥇

**Pull request merged**